### PR TITLE
accept bid dialog

### LINKED ISSUE #1936: Accept an open auction bid with mocked data
As an owner of NFT I want to accept a (open) auction bid.

**AC:**

- [ ] Add button to review bids
- [ ] Add modal with list of bids and radio buttons

**Out of scope:**

- [ ] Fetching bids
- [ ] Handling transaction

###Figma
https://www.figma.com/file/sMydmZJlzhK91FTIPOd7D2/Video-page?node-id=1247%3A100057

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #1936 

**Design**
https://www.figma.com/file/sMydmZJlzhK91FTIPOd7D2/Video-page?node-id=1247%3A100057

### CODE REVIEW

File: packages/atlas/src/components/_icons/JoyTokenIcon/JoyTokenIcon.tsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ export type JoyTokenIconProps = {
   variant?: JoyTokenIconVariant
   size?: JoyTokenIconSize
   className?: string
+  hasShadow?: boolean
```
Comment (kdembler) [inline packages/atlas/src/components/_icons/JoyTokenIcon/JoyTokenIcon.tsx]: Like mentioned in rocket chat, let's remove this prop and use monochrome version instead, just with a different color applied

**Review by drillprop — APPROVED** [review]
Looks good to me.

**Comment by rafalpawlow** [comment]
@drillprop I need to ask for review it again. We will have more modals like this and I decided to move it to separate component. Let me know what you think.

**Review by drillprop — CHANGES_REQUESTED** [review]
AllNftDialogs shouldn't be placed in NFTWidget, dialogs could be opened also from VideoTile. After a short discussion on rocket chat, we decided that we will need to add a provider, which will allow using NFT dialogs in other places.

File: packages/atlas/src/providers/nftDialogs.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import { AcceptBidDialog } from '@/components/_overlays/AcceptBidDialog'
+
+type Dialogs = 'accept-bid' | 'change-price' | 'sign-in'
+
+type ContextValue = {
+  setOpenedDialog: React.Dispatch<React.SetStateAction<Dialogs | undefined>>
+}
+
+export const NftDialogsContext = React.createContext<ContextValue | undefined>(undefined)
```
Comment (kdembler) [inline packages/atlas/src/providers/nftDialogs.tsx]: Any reason we need a separate context? Couldn't we use the one we already have for NFT actions?

File: packages/atlas/src/components/_overlays/AcceptBidDialog/AcceptBidList.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+type BidRowProps = {
+  id: string
+  memberHandle: string
+  date: Date
+  bid: number
+  bidUSD: string
+  memberAvatarUri: string
+  selectedValue?: string
+  handleSelect?: (selectedBid: string) => void
```
Comment (kdembler) [inline packages/atlas/src/components/_overlays/AcceptBidDialog/AcceptBidList.tsx]: `on*` for props, `handle` for handlers

Comment (rafalpawlow) [reply packages/atlas/src/components/_overlays/AcceptBidDialog/AcceptBidList.tsx]: $#@%!!!

File: packages/atlas/src/providers/nftActions/provider.tsx [hunk 4]
```diff
@@ -38,15 +40,25 @@ export const NftActionsProvider: React.FC = ({ children }) => {
   }, [cachedCurrentAction, currentAction, decrementOverlaysOpenCount, incrementOverlaysOpenCount])
   // --END--
```
Comment (kdembler) [inline packages/atlas/src/providers/nftActions/provider.tsx:41]: Leftover from old code

```suggestion
```

File: packages/atlas/src/components/_overlays/AcceptBidDialog/AcceptBidList.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  return (
+    <BidRowWrapper selected={selected}>
+      <RadioInput onChange={() => handleSelect?.(id)} selectedValue={selectedValue} value={id} />
+      {xsMatch && <Avatar assetUrl={memberAvatarUri} size="small" />}
+      <div>
+        <Text variant="h300" margin={{ bottom: 1 }} secondary={!selected}>
+          {memberHandle}
+        </Text>
+        <Text secondary variant="t100">
+          {format(date, "LLL dd yyy 'at' HH:mm")}
```
Comment (kdembler) [inline packages/atlas/src/components/_overlays/AcceptBidDialog/AcceptBidList.tsx]: While you correctly implemented the design here, it seems this specific design doesn't properly follow the guidelines we've set for dates and times. I made a comment to sort this out, in the meantime I suggest we use `formatDateTime` from our date utils here instaed

File: packages/atlas/src/components/_overlays/AcceptBidDialog/AcceptDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+import { useTokenPrice } from '@/providers/joystream'
+
+import { AcceptBidList } from './AcceptBidList'
+
+type AcceptBidDialogProps = {
+  onModalClose: () => void
+  isOpen: boolean
+}
+
+export const AcceptBidDialog: React.FC<AcceptBidDialogProps> = ({ onModalClose, isOpen }) => {
```
Comment (kdembler) [inline packages/atlas/src/components/_overlays/AcceptBidDialog/AcceptDialog.tsx]: The file should be named the same as the component

**Review by kdembler — CHANGES_REQUESTED** [review]
There's a couple of styling issues:
1. The padding for bid rows is incorrect across breakpoints.
2. Same for sizing, in Figma they are 60 or 68px high, that's not the case in the implementation
3. I think the full row should be clickable, not only the checkbox. At the moment it's a bit hard to select a bid
4. The colors for hovered/pressed state are not correct

File: packages/atlas/src/components/_nft/NftWidget/NftWidget.tsx [hunk 7]
```diff
@@ -435,12 +437,23 @@ export const NftWidget: React.FC<NftWidgetProps> = ({
...
+    needsSettling,
+    ownerAvatarUri,
+    onNftAcceptBid,
+  ])
 
   if (!nftStatus) return null
 
   return (
     <Container ref={ref}>
+      <Button onClick={onNftAcceptBid}>Test</Button>
```
Comment (drillprop) [inline packages/atlas/src/components/_nft/NftWidget/NftWidget.tsx]: I think we can remove this button, right? If we want to test something, we can just create a playground for it.

File: packages/atlas/src/components/_overlays/AcceptBidDialog/AcceptBidList.styles.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  justify-content: flex-end;
+  margin-bottom: ${sizes(1)};
+`
+
+export const Date = styled(Text)`
+  display: block;
+`
+
+export const PriceUsd = styled(Text)`
+  display: block;
```
Comment (drillprop) [inline packages/atlas/src/components/_overlays/AcceptBidDialog/AcceptBidList.styles.ts]: Correct me if I'm wrong, but I think we don't need this. We could just use `<Text as="p(or div)" />`.

File: packages/atlas/src/providers/nftActions/provider.tsx [hunk 9]
```diff
@@ -8,19 +9,18 @@ type ContextValue = {
...
 
-type NftAction = 'putOnSale' | 'purchase' | 'settle'
+type NftAction = 'putOnSale' | 'purchase' | 'settle' | 'accept-bid' | 'change-price'
 
 export const NftActionsProvider: React.FC = ({ children }) => {
   const [currentAction, setCurrentAction] = useState<NftAction | null>(null)
   const [currentNftId, setCurrentNftId] = useState<string | null>(null)
 
-  // TODO: remove following code once NftPurchaseView uses BottomDrawer
-  // --START--
```
Comment (drillprop) [inline packages/atlas/src/providers/nftActions/provider.tsx]: Wait, we shouldn't remove this. This block(with useEffect and state) will be removed in #2310.

File: packages/atlas/src/providers/nftActions/provider.tsx [hunk 10]
```diff
@@ -36,17 +36,26 @@ export const NftActionsProvider: React.FC = ({ children }) => {
       decrementOverlaysOpenCount()
     }
   }, [cachedCurrentAction, currentAction, decrementOverlaysOpenCount, incrementOverlaysOpenCount])
-  // --END--
```
Comment (drillprop) [inline packages/atlas/src/providers/nftActions/provider.tsx:39]: Same as above - I would keep these comments.

**Review by drillprop — CHANGES_REQUESTED** [review]

File: packages/atlas/src/components/_nft/NftWidget/NftWidget.tsx (hunk 7 above)
Comment (rafalpawlow) [reply packages/atlas/src/components/_nft/NftWidget/NftWidget.tsx]: This is unnecessary, I forget to remove it

File: packages/atlas/src/providers/nftActions/provider.tsx (hunk 10 above)
Comment (rafalpawlow) [reply packages/atlas/src/providers/nftActions/provider.tsx:39]: Actually, @kdembler  asked me to remove this https://github.com/Joystream/atlas/pull/2307#discussion_r832169316

Comment (kdembler) [reply packages/atlas/src/providers/nftActions/provider.tsx:39]: Ahh my bad, I thought the PR where this was removed was merged already! let's keep it then!

**Review by kdembler — APPROVED** [review]

**Review by drillprop — COMMENTED** [review]
Needs rebase

**Review by drillprop — APPROVED** [review]

**Pull request merged**